### PR TITLE
doc: Added information on --indent to --help

### PR DESCRIPTION
**Pull request by EvanCarroll**
Simple doc patch when I noticed functionality I already need was there just undocumented.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -81,6 +81,8 @@ static void usage(int code, int keep_it_short) {
       "  -M               monochrome (don't colorize JSON);\n"
       "  -S               sort keys of objects on output;\n"
       "  --tab            use tabs for indentation;\n"
+      "  --indent         use indent amount of spaces for indentation.\n"
```
Comment (wader) [inline src/main.c:84]: Maybe should be `--indent n` to be clearer?

Comment (EvanCarroll) [reply src/main.c]: @wader fixed

Comment (wader) [reply src/main.c]: 👍 hopefully will get merged eventually, see #2305 for details

**Comment by pkoppstein** [comment]
Since --indent requires an argument, an argument should be mentioned in usage(). Following the manual, `--indent n` would be appropriate.

The larger point here is that documenting all the options in usage() in this manner would increase the size of the executable quite a lot, and goes against the idea of keeping the executable small by providing details online.

I would propose that usage() simply provide a synopsis of *all* the options (in the usual man page style), leaving the descriptive details of the options to the online documentation.

**Comment by EvanCarroll** [comment]
@pkoppstein the patch was already amended 12 hours before your comment to include `n`. And you're talking about a couple of bytes of text. Hardly something that will "increase the size of the executable quite a lot."

**Comment by pkoppstein** [comment]
@EvanCarroll - You apparently overlooked the phrase “ … documenting all the options in usage() …”.

(The screen I was looking at only showed the line without the “n”.)

**Comment by wader** [comment]
I would not worry about binary size either, possibly that the text is getting long, but maybe then we should cut down on some other usage text instead? maybe the stuff about IEEE754 numbers etc

**Review by itchyny — COMMENTED** [review]
I'm not worried about the executable size due to this help, we should list all the option #2284 (we should rather consider replace the builtins by C-coded internals). The only concern about this PR is about the newline in the description; can Bash's `_parse_help` parse this form?

**Comment by itchyny** [comment]
Looks OK.
```sh
 $ _parse_help jq -h | xargs
-c -n -e -s -r -R -C -M -S --tab --arg --argjson --slurpfile --rawfile --args --jsonargs --
 $ _parse_help ./jq -h | xargs
-c -n -e -s -r -R -C -M -S --tab --indent --arg --argjson --slurpfile --rawfile --args --jsonargs --
```

**Review by itchyny — APPROVED** [review]

File: src/main.c [hunk 2]
```diff
@@ -81,6 +81,8 @@ static void usage(int code, int keep_it_short) {
       "  -M               monochrome (don't colorize JSON);\n"
       "  -S               sort keys of objects on output;\n"
       "  --tab            use tabs for indentation;\n"
+      "  --indent n       use $n spaces for indentation.\n"
+      "                   between -1 (default of 4 spaces) and 7;\n"
```
Comment (itchyny) [inline src/main.c:85]: Oops, `-1` is not `4 spaces`, but the tab character.

**Review by itchyny — CHANGES_REQUESTED** [review]

**Pull request closed without merging**